cli/azd/pkg/project/service_target_aks.go

```diff
@@ -0,0 +1,444 @@
+package project
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"net/url"
+	"path/filepath"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/azure"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment/azdcontext"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/azcli"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/docker"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/kubectl"
+	"github.com/benbjohnson/clock"
+)
+
+const (
+	defaultDeploymentPath = "manifests"
+)
+
+type AksOptions struct {
+	Namespace      string               `yaml:"namespace"`
+	DeploymentPath string               `yaml:"deploymentPath"`
+	Ingress        AksIngressOptions    `yaml:"ingress"`
+	Deployment     AksDeploymentOptions `yaml:"deployment"`
+	Service        AksServiceOptions    `yaml:"service"`
+}
+
+type AksIngressOptions struct {
+	Name         string `yaml:"name"`
+	RelativePath string `yaml:"relativePath"`
+}
+
+type AksDeploymentOptions struct {
+	Name string `yaml:"name"`
+}
+
+type AksServiceOptions struct {
+	Name string `yaml:"name"`
+}
+
+type aksTarget struct {
+	config           *ServiceConfig
+	env              *environment.Environment
+	scope            *environment.TargetResource
+	containerService azcli.ContainerServiceClient
+	az               azcli.AzCli
+	docker           docker.Docker
+	kubectl          kubectl.KubectlCli
+	clock            clock.Clock
+}
+
+func (t *aksTarget) RequiredExternalTools() []tools.ExternalTool {
+	return []tools.ExternalTool{t.docker}
+}
+
+func (t *aksTarget) Deploy(
```

Comment:
@wbreza I think you might have covered a lot of these in the product demo, but it would be great if we document a simple spec that describes how `k8s` deploy is handled in `azd`, the sequencing of steps and the end result.

The questions I'd be most interested in:
1. How does `azd` initiated `k8s`s deployment flow differ from a normal `k8s` deployment flow?
2. The `ingress` handling and how it relates to the healthcheck/wait deployment flow

Reply:
Answered this in teams chat. Let me know if you want to follow-up.